src/main/java/cz/centrum/haffner/SimpleTrainingDavid/AppServices/MetricsInfoService.java

```diff
@@ -1,26 +1,43 @@
 package cz.centrum.haffner.SimpleTrainingDavid.AppServices;
 
 import cz.centrum.haffner.SimpleTrainingDavid.DataTemplates.MetricsInfoData;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
-import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
 
 // algorithm for returning of the proper MetricsInfo data
 @Service
 public class MetricsInfoService {
-    // private static final String FILE_PATH = "https://github.com/DavidHaffner/SimpleTrainingDavid/tree/dev/logs/MCP_20180131.json";
-    private static final String FILE_PATH_PREFIX = MetricsInfoService.class.getClassLoader()
-                                                        .getResource("MCPData/").getPath();
+    private static final Logger logger = LogManager.getLogger(MetricsInfoService.class);
+
+    private static final String FILE_PATH_PREFIX = "https://raw.githubusercontent.com/TomasStesti/simpleTraining/master/logs/";
+    //private static final String FILE_PATH_PREFIX = MetricsInfoService.class.getResource("/MCPData/").getPath();
     @Autowired
-    private OneDayFileJsonParser oneDayFileJsonsParser;
+    private Parser oneDayFileJsonsParser;
+
 
+    public MetricsInfoData processData(long requestedDate) {
+        if(logger.isDebugEnabled()) {
+            logger.debug("Starting to process metrics data of date: " + requestedDate);
+        }
 
-    public MetricsInfoData processMetricsInfoData(long requestedDate) {
         // get the file of requested date
         String filePath = FILE_PATH_PREFIX + "MCP_" + requestedDate + ".json";
-        File oneDayFile = new File(filePath);
 
-        return oneDayFileJsonsParser.process(oneDayFile);
+        URL oneDayFile = null;
+        try {
+            oneDayFile = new URL(filePath);
+            if(logger.isDebugEnabled()) {
+                logger.debug("Successfully received file from: " + filePath);
+            }
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```

Comment:
Pokud se nepodaří úspěšně najít a otevřít soubor, tak se tady vypíše exception do STDOUT (což není chybový výstup) a program směle pokračuje dál - předá null do parse() metody...
Ty vyjímky se odchytávají z nějakého důvodu - Myslíš, že jen vypsat stacktrace a poslat null dál do programu je tady ok?
Nebo by to mělo být vyřešeno jinak?

Reply:
Pardon, chyba -> předělám, aby loggovalo na úrovni ERROR do logu + samostatného error logu.

Reply:
... a přidám řešení do catch bloku, když se nepodaří  soubor korektně načíst (aby algoritmus nepokračoval dál)